ocean-client/src/programs/vault-maxi-program.ts

```diff
@@ -52,17 +60,27 @@ export class VaultMaxiProgram extends CommonProgram {
         const tokens = await this.getTokenBalances()
         console.log(" removed liq. got tokens: " + Array.from(tokens.values()).map(value => " " + value.amount + "@" + value.symbol))
         let paybackTokens: TokenBalance[] = []
+        let tokenIdToSymbol = new Map<Number, string>()
         let token = tokens.get("DUSD")
-        if (token) paybackTokens.push({ token: +token.id, amount: new BigNumber(Math.min(+token.amount, wantedusd)) })
+        if (token) {
+            tokenIdToSymbol.set(+token.id, token.symbol)
+            paybackTokens.push({ token: +token.id, amount: new BigNumber(Math.min(+token.amount, wantedusd)) })
+        }
         token = tokens.get(this.settings.LMToken)
-        if (token) paybackTokens.push({ token: +token.id, amount: new BigNumber(Math.min(+token.amount, neededStock)) })
+        if (token) {   
+            tokenIdToSymbol.set(+token.id, token.symbol)
+            paybackTokens.push({ token: +token.id, amount: new BigNumber(Math.min(+token.amount, neededStock)) })
+        }
 
+        return await this.paybackTokenBalances(paybackTokens, tokenIdToSymbol, telegram)
+    }
 
-        console.log(" paying back tokens " + paybackTokens.map(token => " " + token.amount + "@" + token.token))
+    private async paybackTokenBalances(paybackTokens: TokenBalance[], tokenIdToSymbol: Map<Number, string>, telegram: Telegram): Promise<boolean> {
```

Comment:
hmm if we add another parameter and all: why not change the input to an Array of AddressToken and set the amount in there. would include id and symbol, no artifical map (just for logging) and can retrieve the TokenBalances for the api call in here then.

Reply:
I had that, changed it, because I decided in a different way how much should be paid back. But with my last comment - I made the suggestion to go for `min(token.amount, loan.amount)` which basically nullify this change.